Jenkinsfile

```diff
@@ -49,16 +40,29 @@ pipeline {
         sh 'git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master'
         sh 'git fetch --no-tags'
 
-        // Don't fail the whole build if octocatalog-diff fails, since it's new
-        // and needs some fixing before it's relied on
         script {
-          try {
-            sh 'make all_diffs'
-          } catch (err) {
-            echo 'make all_diffs failed, but it is being ignored for now'
-            mail to: 'jvperrin@ocf.berkeley.edu',
-                 subject: "all_diffs failed on ${JOB_NAME}/#${BUILD_NUMBER}",
-                 body: BUILD_URL
+          // This should only run for pull requests, so that it is able to post
+          // change/failure comments on the review
+          if (env.CHANGE_ID) {
+            // Unfortunately both the output and the status code cannot be
+            // saved at the same time (thanks jenkins, see
+            // https://issues.jenkins-ci.org/browse/JENKINS-44930), so the
+            // output is saved to a file and then used soon after
+            def status = sh returnStatus: true, script: 'make all_diffs > all_diffs_output.log'
+            // GitHub has a max comment length of 65536, so truncate and leave
```

Comment:
perhaps we could post extra long diffs to fluffy or something? Or perhaps a github gist? (Note I don't think this has to be done immediately, just a follow up idea).

Reply:
Hmmm, I do like that, it would be nicer than telling people to just go look at the jenkins log...

I especially like the idea of making a gist out of the results, as then it would have the same collapsable sections, where a fluffy link would not unless each error or diff is uploaded separately or something and then linked.

Reply:
Ok, I changed this to post to a gist instead, and opened https://github.com/ocf/shared-pipeline/pull/7 to add that functionality to jenkins.